cmd/gaiacli/README.md

```diff
@@ -0,0 +1,148 @@
+
+```shell
+# Configure the proper database backend for each node
+sed -i '' 's/"leveldb"/"goleveldb"/g' ibc0/n0/gaiad/config/config.toml
+sed -i '' 's/"leveldb"/"goleveldb"/g' ibc1/n0/gaiad/config/config.toml
+
+# Configure chain ibc1 to have different listening ports
+sed -i '' 's#"tcp://0.0.0.0:26656"#"tcp://0.0.0.0:26556"#g' ibc1/n0/gaiad/config/config.toml
+sed -i '' 's#"tcp://0.0.0.0:26657"#"tcp://0.0.0.0:26557"#g' ibc1/n0/gaiad/config/config.toml
+sed -i '' 's#"localhost:6060"#"localhost:6061"#g' ibc1/n0/gaiad/config/config.toml
+sed -i '' 's#"tcp://127.0.0.1:26658"#"tcp://127.0.0.1:26558"#g' ibc1/n0/gaiad/config/config.toml
+```
+
+Then configure your `gaiacli` instances for each chain:
+
+```bash
+gaiacli config --home ibc0/n0/gaiacli/ chain-id ibc0
+gaiacli config --home ibc1/n0/gaiacli/ chain-id ibc1
+gaiacli config --home ibc0/n0/gaiacli/ node http://localhost:26657
+gaiacli config --home ibc1/n0/gaiacli/ node http://localhost:26557
+
+# Add the key from ibc1 to the ibc0 cli
+jq -r '.secret' ibc1/n0/gaiacli/key_seed.json | pbcopy
+
+# Paste the mnemonic from the above command after setting password (12345678)
+gaiacli --home ibc0/n0/gaiacli keys add n1 --recover
+```
+
+After configuration is complete, start each node in a seperate terminal window:
+
+```bash
+gaiad --home ibc0/n0/gaiad start
+gaiad --home ibc1/n0/gaiad start
+```
+
+## Client
+
+Create a client on ibc1:
+
+```bash
+gaiacli --home ibc0/n0/gaiacli q ibc client path > path0.json
+gaiacli --home ibc0/n0/gaiacli q ibc client consensus-state > state0.json
+gaiacli --home ibc1/n0/gaiacli tx ibc client create c1 ./state0.json --from n0
+gaiacli --home ibc1/n0/gaiacli q ibc client client c1
+```
+
+Create a client on ibc0:
+
+```bash
+gaiacli --home ibc1/n0/gaiacli q ibc client path > path1.json
+gaiacli --home ibc1/n0/gaiacli q ibc client consensus-state > state1.json
+gaiacli --home ibc0/n0/gaiacli tx ibc client create c0 ./state1.json --from n0
+gaiacli --home ibc0/n0/gaiacli q ibc client client c0
+```
+
+## Connection
+
+Connections can be established with `connection.sh $CLIENTID` command. It will print
+
+```shell
+gaiacli \
+  --home ibc0/n0/gaiacli \
+  tx ibc connection handshake \
+  conn0 c0 path1.json \
+  conn1 c1 path0.json \
+  --from1 n0 --from2 n1 \
+  --node1 tcp://localhost:26657 \
+  --node2 tcp://localhost:26557
+```
+
+Once the connection is established you should be able to query it:
+
+```bash
+gaiacli --home ibc0/n0/gaiacli q ibc connection connection conn0 --trust-node
+```
+
+## Channel
+
+To establish a channel using the `ibc-mock` application protocol run the following command:
+
+```
+gaiacli \
+  --home ibc0/n0/gaiacli \
+  tx ibc channel handshake \
+  ibc-mock chan0 conn0 \
+  ibc-mock chan1 conn1 \
+  --from1 n0 --from2 n1
+```
+
+You can query the channel after establishment by
+
+```bash
+gaiacli --home ibc0/n0/gaiacli query ibc channel channel ibc-mock chan0 --trust-node
+```
+
+## Send Packet
```

Comment:
Both send-packet commands are using `ibc0`, but all the other sections have matched pairs of `ibc0`/`ibc1` commands. Is this intentional? I'm guessing it's just a typo.

Reply:
Yes likely. That piece of the instructions is still from the loopback demo. What do you think it should look like?

Reply:
How should the commands look?

Reply:
`gaiacli tx ibc channel flush` is currently uses origin chain as main chain and receiving chain as side chain, which is the reason it is using `ibc0` as home, bit misleading

Reply:
`flush` command is currently using `ibc0` as its home but making transaction on `ibc1`, which is bit misleading

Reply:
@mossid so how would you rewrite the example?